Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateParser.cs

```diff
@@ -284,8 +288,98 @@ private DTParseResult ParseNumberWithMonth(string text, DateObject referenceDate
             ret.Success = true;
 
             return ret;
-        }
-
+        }
+
+        // handle like "two days ago" 
+        private DTParseResult ParserDurationWithAgoAndLater(string text, DateObject referenceDate)
+        {
+            var ret = new DTParseResult();
+            var duration_res = config.DurationExtractor.Extract(text);
+            var numStr = string.Empty;
+            var unitStr = string.Empty;
+            if (duration_res.Count > 0)
+            {
+                var match = this.config.UnitRegex.Match(text);
+                if (match.Success)
+                {
+                    var AfterStr =
+                        text.Substring((int)duration_res[0].Start + (int)duration_res[0].Length)
+                            .Trim()
+                            .ToLowerInvariant();
+                    var srcUnit = match.Groups["unit"].Value.ToLowerInvariant();
+                    var numberStr =
+                        text.Substring((int)duration_res[0].Start, match.Index - (int)duration_res[0].Start)
+                            .Trim()
+                            .ToLowerInvariant();
+                    var er = config.CardinalExtractor.Extract(numberStr);
+                    if (er.Count != 0)
+                    {
+                        var pr = config.NumberParser.Parse(er[0]);
+
+                        var number = int.Parse(pr.ResolutionStr);
+                        if (config.UnitMap.ContainsKey(srcUnit))
+                        {
+                            unitStr = config.UnitMap[srcUnit];
+                            numStr = number.ToString();
+                            if (AfterStr.StartsWith("ago"))

```

Comment:
Same as above. Code could also be generalized to just pass the sign (past, future) to a common method.